codegen/input.gotpl

```diff
@@ -2,7 +2,10 @@
 	{{- if not .HasUnmarshal }}
 	func (ec *executionContext) unmarshalInput{{ .Name }}(ctx context.Context, obj interface{}) ({{.Type | ref}}, error) {
 		var it {{.Type | ref}}
-		var asMap = obj.(map[string]interface{})
+		asMap := map[string]interface{}{}
+		for k, v := range obj.(map[string]interface{}) {
```

Comment:
Does this copy need to go deeper for nested types?

Reply:
@lwc  Thanks for the review.

gqlgen will set the default values for fields (i.e., asMap's direct values).
So I don't think we need deep copy here. 

```
{{ range $field := .Fields}}
	{{- if $field.Default}}
		if _, present := asMap[{{$field.Name|quote}}] ; !present {
			asMap[{{$field.Name|quote}}] = {{ $field.Default | dump }}
		}
	{{- end}}
{{- end }}
```